src/doubly_linked_list.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding utf-8 -*-
+
+
+class Node(object):
+
+    def __init__(self, data=None, pre_node=None, next_node=None):
+        self.pre_node = pre_node
+        self.data = data
+        self.next_node = next_node
+
+
+class DoubleLinkedList(object):
+
+    def __init__(self, iterable=None):
+        self.head = None
+        self.tail = None
+        if iterable is not None:
+            for i in iterable:
+                self.push(i)
+
+    def push(self, val):
+        ''' Push a node onto the head of the list '''
+        new_node = Node(val)
```

Comment:
These blocks of code work great, and there is also room for improvement. Every line of code is a potential liability, so think about where you can reduce lines of code here.